Sources/TuistCore/Graph/Mappers/GraphMapper.swift

```diff
@@ -5,21 +5,21 @@ import Foundation
 public protocol GraphMapping {
     /// Given a graph, it maps it into another graph.
     /// - Parameter graph: Graph to be mapped.
-    func map(graph: Graph) throws -> (Graph, [SideEffectDescriptor])
+    func map(graph: Graph) throws -> (Graph, Set<SideEffectDescriptor>)
```

Comment:
The order in which side effects are performed matters

e.g.

- delete derived directory
- create derived directory
- create file

Placing those in a set will not guarantee the correct execution of steps :/

Reply:
Ouch, I missed that one 🤦